src/attribute_type.rs

```diff
@@ -2327,67 +2808,89 @@ fn test_parse_attribute_group() {
+                                                                    SdpAttribute::IcePwd);
+
+    check_parse_and_serialize("ice-pwd:e3baa26dd2fa5030d881d385f1e36cce");
 
     assert!(parse_attribute("ice-pwd:").is_err());
 }
 
 #[test]
 fn test_parse_attribute_ice_ufrag() {
-    assert!(parse_attribute("ice-ufrag:58b99ead").is_ok());
+    let check_parse = make_check_parse!(String, SdpAttribute::IceUfrag);
+    let check_parse_and_serialize = make_check_parse_and_serialize!(check_parse,
+                                                                    SdpAttribute::IceUfrag);
+
+    check_parse_and_serialize("ice-ufrag:58b99ead");
 
     assert!(parse_attribute("ice-ufrag:").is_err());
 }
 
 #[test]
 fn test_parse_attribute_identity() {
-    assert!(parse_attribute("identity:eyJpZHAiOnsiZG9tYWluIjoiZXhhbXBsZS5vcmciLCJwcm90b2NvbCI6ImJvZ3VzIn0sImFzc2VydGlvbiI6IntcImlkZW50aXR5XCI6XCJib2JAZXhhbXBsZS5vcmdcIixcImNvbnRlbnRzXCI6XCJhYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3l6XCIsXCJzaWduYXR1cmVcIjpcIjAxMDIwMzA0MDUwNlwifSJ9").is_ok());
+    let check_parse = make_check_parse!(String, SdpAttribute::Identity);
+    let check_parse_and_serialize = make_check_parse_and_serialize!(check_parse,
+                                                                    SdpAttribute::Identity);
+
+    check_parse_and_serialize("identity:eyJpZHAiOnsiZG9tYWluIjoiZXhhbXBsZS5vcmciLCJwcm90b2NvbCI6ImJvZ3VzIn0sImFzc2VydGlvbiI6IntcImlkZW50aXR5XCI6XCJib2JAZXhhbXBsZS5vcmdcIixcImNvbnRlbnRzXCI6XCJhYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3l6XCIsXCJzaWduYXR1cmVcIjpcIjAxMDIwMzA0MDUwNlwifSJ9");
 
     assert!(parse_attribute("identity:").is_err());
 }
 
 #[test]
 fn test_parse_attribute_imageattr() {
-    let check_parse = |x| -> SdpAttributeImageAttr {
-        if let Ok(SdpType::Attribute(SdpAttribute::ImageAttr(x))) = parse_attribute(x) {
-            x
-        } else {
-            unreachable!();
-        }
-    };
-
-    assert!(parse_attribute("imageattr:120 send * recv *").is_ok());
-    assert!(parse_attribute("imageattr:99 send [x=320,y=240] recv [x=320,y=240]").is_ok());
-    assert!(parse_attribute("imageattr:97 send [x=800,y=640,sar=1.1,q=0.6] [x=480,y=320] recv [x=330,y=250]").is_ok());
+    let check_parse = make_check_parse!(SdpAttributeImageAttr, SdpAttribute::ImageAttr);
+    let check_parse_and_serialize = make_check_parse_and_serialize!(check_parse,
+                                                                    SdpAttribute::ImageAttr);
+
+    check_parse_and_serialize("imageattr:120 send * recv *");
+    check_parse_and_serialize("imageattr:99 send [x=320,y=240] recv [x=320,y=240]");
+    check_parse_and_serialize("imageattr:97 send [x=800,y=640,sar=1.1,q=0.6] [x=480,y=320] recv [x=330,y=250]");
+    check_parse_and_serialize("imageattr:97 send [x=[480:16:800],y=[320:16:640],par=[1.2-1.3],q=0.6] [x=[176:8:208],y=[144:8:176],par=[1.2-1.3]] recv *");
     assert!(parse_attribute("imageattr:97 recv [x=800,y=640,sar=1.1] send [x=330,y=250]").is_ok());
```

Comment:
Because we only check the parsing at this point. The serialization will not be equal as the serialization will always serialize "send" first and "recv" afterwards.